cloudmock/openstack/mocknetworking/floatingips.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mocknetworking
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/floatingips"
+)
+
+type floatingIPListResponse struct {
+	FloatingIPs []floatingips.FloatingIP `json:"floatingips"`
+}
+
+func (m *MockClient) mockFloatingIPs() {
+	re := regexp.MustCompile(`/floatingips/?`)
+
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		m.mutex.Lock()
+		defer m.mutex.Unlock()
+
+		w.Header().Add("Content-Type", "application/json")
+
+		floatingIPID := re.ReplaceAllString(r.URL.Path, "")
+		switch r.Method {
+		case http.MethodGet:
+			if floatingIPID == "" {
+				r.ParseForm()
+				m.listFloatingIPs(w, r.Form)
```

Comment:
This mock of floating IPs only has "list" support. How do they get created? does kops create them (in which case we need to look for create requests) or does the creation of another resource cascade to floating IPs? Or are they expected to already exist, in which case we should either create them in integrationtestharness.go or create them in mockFloatingIPs like I did for availability zones

Reply:
The floating IP is implicitly created when the port is added to the external network, I believe.

There is something similar with routerinterface too. A router interface is just a port that is implicitly created when calling add_router_interface. Looks like openstack has a lot of these implicit creations.

Reply:
I also see them being created here: https://github.com/kubernetes/kops/blob/fefe747f83e4c5d9fec331e6e5fb244e9b184cf0/upup/pkg/fi/cloudup/openstacktasks/floatingip.go#L210-L236

so this might need to do support creation requests if this kops code is exercised in our integration tests

Reply:
Yep. We need to add more integration tests that exercise other configurations.